lib/src/widgets/yaru_page_indicator.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:flutter/material.dart';
+
+/// A Yaru page indicator.
+///
+/// If there's enough space, it will be rendered into a line of dots,
+/// if not, it will be rendered into a text-based indicator.
+class YaruPageIndicator extends StatelessWidget {
+  /// Create a [YaruPageIndicator].
+  const YaruPageIndicator({
+    super.key,
+    required this.length,
+    required this.page,
+    this.animationDuration = Duration.zero,
+    this.animationCurve = Curves.linear,
+    this.onDotTap,
+  }) : assert(page >= 0 && page <= length - 1);
+
+  /// Determine the number of pages.
+  final int length;
+
+  /// Current page index.
+  /// This value should be clamped between 0 and [length] - 1
+  final int page;
```

Comment:
could it be called "index" to follow the same naming as YaruPageController?

- https://pub.dev/documentation/yaru_widgets/latest/yaru_widgets/YaruPageController/index.html

Reply:
Well, I used `page` because `PageController`, also`YaruCarouselController` who extends it, and `YaruCarousel` are using this name everywhere.

If I rename it here, should I rename it everywhere?

Reply:
No strong opinion on this :)